File: unityPart/Assets/Scripts/Controller/CameraController.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using System;
using Base;
using XLua;

public class CameraController : IController
{
    //GameObject avatar;
    //GameObject camera;

    //float RotateScale = 8f;

    ///// <summary>
    ///// 范围
    ///// </summary>
    //Vector2 Vertical = new Vector2(0, 60); 

    [CSharpCallLua]
    public delegate Action LuaAwake();
    [CSharpCallLua]
    public delegate Action LuaStart();
    [CSharpCallLua]
    public delegate Action LuaUpdate();

    public void Awake()
    {
        Debug.Log("1231232131");

    }

    // Start is called before the first frame update
    public void Start()
    {
    }

    // Update is called once per frame
    public void Update()
    {
        //Vector3 centerPos = avatar.transform.position;
        ///////水平旋转
        //camera.transform.RotateAround(centerPos, Vector3.up, Input.GetAxisRaw("Mouse X") * Time.timeScale * RotateScale);

        ///////垂直旋转
        //Vector3 oldPos = camera.transform.position;
        //Quaternion oldRatation = camera.transform.rotation;
        //camera.transform.RotateAround(centerPos, camera.transform.right, Input.GetAxisRaw("Mouse Y") * Time.timeScale * RotateScale / 4);
        //if(camera.transform.rotation.eulerAngles.x < Vertical.x || camera.transform.rotation.eulerAngles.x > Vertical.y)
        //{
        //    camera.transform.position = oldPos;
        //    camera.transform.rotation = oldRatation;
        //}
    }
}

File: unityPart/Assets/Scripts/Lua/Controllers/CameraController.lua
-- local Vector3 = CS.UnityEngine.Vector3;

-- local private = {
--     RotateScale = 8.0,
--     VerticalRange = {
--         min = 0,
--         max = 60
--     }
-- }
-- local public  = {}



-- public.Awake = function()
    
-- end

-- public.SetAvatar = function(avatar)
--     private.avatar = avatar
--     private.camera.transform.position =  Vector3(private.avatar.transform.position.x,private.avatar.transform.position.y + 2,private.avatar.transform.position.z-2.5)
-- end

-- public.Start = function()
--     private.avatar = nil
--     private.camera = Global.mainCamera
--     private.LogicManager = Global.Managers.LogicManager
-- end

-- public.Update = function()
--     if(private.avatar ~= nil and private.LogicManager.isBattle) then
--         local centerPos = private.avatar.transform.position;
--         private.camera.transform:RotateAround(centerPos, Vector3.up, CS.UnityEngine.Input.GetAxisRaw("Mouse X") * CS.UnityEngine.Time.timeScale * private.RotateScale)
--         local oldPos = private.camera.transform.position
--         local oldRatation = private.camera.transform.rotation
--         private.camera.transform:RotateAround(centerPos, private.camera.transform.right, CS.UnityEngine.Input.GetAxisRaw("Mouse Y") * CS.UnityEngine.Time.timeScale * private.RotateScale / 4)
--         if(private.camera.transform.rotation.eulerAngles.x < private.VerticalRange.min or private.camera.transform.rotation.eulerAngles.x > private.VerticalRange.max) then
--             private.camera.transform.position = oldPos;
--             private.camera.transform.rotation = oldRatation;
--         end
--     end
-- end

-- return public